src/transformers/generation/utils.py

```diff
@@ -1421,6 +1421,11 @@ def generate(
                 )
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
 
+        # adjust max_length when using `input_embeds` in decoder-only models
+        elif len(inputs_tensor.shape) == 3 and inputs_tensor.shape[:-1] != input_ids.shape:
```

Comment:
```suggestion
        elif model_input_name == "inputs_embeds" and inputs_tensor.shape[:-1] != input_ids.shape:
```
`len(inputs_tensor.shape) == 3` may clash with other modalities (present or future). `model_input_name == "inputs_embeds"` also implies `len(inputs_tensor.shape) == 3` :D